aten/src/ATen/native/nested/NestedTensorMath.cpp

```diff
@@ -165,7 +165,8 @@ Tensor NestedTensor_nested_tensor_from_mask(const Tensor& t, const Tensor& mask)
     sizes = sizes.cumsum(1).select(1, L - 1);
     nums = nums.to(sizes.options());
 
-    TORCH_CHECK(sizes.equal(nums), "Mask must be left-aligned without gaps");
+    if (mask_check)
+      TORCH_CHECK(sizes.equal(nums), "Mask must be left-aligned without gaps");
```

Comment:
Seems like `nums` is only used for this check. Should we avoid computing it at all when `mask_check=True`?